components/Profile/modal-views/index.tsx

```diff
@@ -449,6 +452,42 @@ export const Questions = ({ handleOk }: { handleOk: () => void }) => {
     );
 };
 
+export const QRCanvas = ({ handleOk }: { handleOk: () => void }) => {
+    const { Canvas } = useQRCode();
+    const [uri, setUri] = useState<String>('');
+    const [isLoading, setIsLoading] = useState(true);
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                getAttendeeQrURI().then((data) => {
+                    setUri(data);
+                    setIsLoading(false);
+                });
+                
+            } catch (error) {
+                console.error('Error fetching QR data:', error);
+            } 
+        };
+        fetchData();
+    }, [isLoading]); 
+
+    if (uri) {
```

Comment:
idk why i had to add this but it wasnt working without it? @nrking0 please look into it if you can